File: Week_07/G20190343020061/SelectionSort.php
<?php

function selectionSort($arr)
{
    $len = count($arr);
    if ($len <= 0) return $arr;
    for ($i = 0; $i < $len - 1; $i++) {
        $momentIndex = $i;
        for ($j = $i + 1; $j < $len; $j++) {
            if ($arr[$j] < $arr[$momentIndex]) {
                $momentIndex = $j;
            }
        }
        $tmp = $arr[$i];
        $arr[$i] = $arr[$momentIndex];
        $arr[$momentIndex] = $tmp;
    }
    return $arr;
}

File: Week_07/G20190343020289/SelectionSort.go
package G20190343020289

// 快速排序
func selectionSort(arr []int) []int {
	length := len(arr)
	minIndex := 0
	for i := 0; i < length-1; i++ {
		minIndex = i
		for j := i + 1; j < length; j++ {
			if arr[j] < arr[minIndex] {
				minIndex = j
			}
		}
		arr[i], arr[minIndex] = arr[minIndex], arr[i]
	}
	return arr
}
